core/src/main/scala/cats/syntax/either.scala

```diff
@@ -189,9 +254,13 @@ final class EitherOps[A, B](val eab: Either[A, B]) extends AnyVal {
    * }}}
    */
   def toEitherT[F[_]: Applicative]: EitherT[F, A, B] = EitherT.fromEither(eab)
+
+  private[cats] def leftCast[C]: Either[C, B] = eab.asInstanceOf[Either[C, B]]
+
+  private[cats] def rightCast[C]: Either[A, C] = eab.asInstanceOf[Either[A, C]]
```

Comment:
Since it's intended to be used only internally perhaps it would be more appropriate to put it on an object somewhere? Like `private[cats] object EitherUtil` (still in `syntax/either.scala` since most usages are there)